js/core/transform-json-response.js

```diff
@@ -1,99 +1,72 @@
-    idAttribute: (a) => `${a.embeddableKey}-${a.studentId}`,
-  });
-  const feedback = new schema.Entity("feedbacks", {}, {
-    idAttribute: (a) => a.answerKey,
-  });
-
-  const activityFeedback = new schema.Entity("activityFeedbacks", {}, {
-    idAttribute: (value, parent) => `${parent.id}-${value.studentId}`,
-  });
-
-  investigation.define({
+  sequence.define({
     children: [ activity ],
   });
   activity.define({
     children: [ section ],
-    activityFeedback: [ activityFeedback ],
   });
   section.define({
     children: [ page ],
   });
   page.define({
     children: [ question ],
   });
-  question.define({
-    answers: [ answer ],
-  });
-  answer.define({
-    feedbacks: [ feedback ],
-  });
+  return normalize(camelizedJson, sequence);
+}
 
-  const reportType = camelizedJson.report.type;
-  const response = normalize(migrate(camelizedJson), {
-    report: reportType === "Investigation" ? investigation : activity,
-    "class": {
-      students: [ student ],
-    },
-  });
-  // Post-process response:
-  // Provide fake investigation if it's not present to simplify app logic.
-  if (reportType === "Activity") {
-    response.entities.investigations = {
-      1: {
-        id: 1,
-        name: "",
-        type: "Investigation",
-        children: [response.result.report],
-      },
+export function preprocessResourceJson(resourceJson) {
+  // Provide fake sequence if it's not present to simplify app logic.
+  if (resourceJson.type === "Activity") {
+    resourceJson = {
+      id: 1,
+      name: "",
+      type: "Sequence",
+      children: [ resourceJson ],
     };
   }
-  response.type = camelizedJson.reportFor || DEFAULT_REPORT_FOR;
-  response.hideControls = camelizedJson.hideControls || false;
-  applyVisibilityFilter(response.entities.questions, response.result.visibilityFilter);
-  copyAnswerKeysToObjects(response.entities.answers || []);
-  copyAnswerKeysToObjects(response.entities.activityFeedbacks || []);
-  saveStudentsRealNames(response.entities.students);
-  urlParamOverrides(response);
-  return response;
+  // Add some question properties, e.g. question numbers, selection, visibility.
+  resourceJson.children.forEach(activity => {
+    activity.children.forEach(section => {
+      // Questions are numbered within the whole activity.
+      let currentQuestionNumber = 1;
+      section.children.forEach(page => {
+        page.children.forEach(question => {
+          question.questionNumber = currentQuestionNumber++;
+          // Id is not enough, as questions of different type can have the same id.
+          question.key = `${question.type}-${question.id}`;
+          // Nothing is selected by default.
+          question.selected = false;
+          if (question.type === "multiple_choice") {
+            // Multiple choice question is scored if at least one choice is marked as correct.
+            question.scored = question.choices.some(c => c.correct);
```

Comment:
Same question here:  Should we export the 'scored' question property as part of the new schema?

Reply:
Having a scored option would enable us to support an multi-select question that had no correct answers. That seems like a good thing. LARA doesn't support this now, but it seems good to make it explicit.

Reply:
Hm, I'm 50-50 here. I feel like `scored` can be always derived from `correct` properties. So LARA in fact already provides this information. This logic here is just simplification for UI components, so they don't have to do this check all the time. 
Is there a case when the question is scored, but none of the choices is considered to be correct?

@scytacki, I'm not sure if I understand your use case and how `scored` would relate to multi-select questions. Note that here we talk about question metadata - whether a given choice is correct. It's not an answer and whether a choice is selected (if that's you meant).

Reply:
For a normal multiple choice question the student has to choose one and only one answer. Once they select a answer there is no way to unselect all of the answers.  So if it is scored, it will always have a correct answer.

For a multiple select multiple choice. The student could choose 0 to N answers. And even if they choose 1 answer they can still unselect it.  So in theory choosing 0 could be the correct way to respond to the question.  For example: "Select all of the items that are true." and then none of the choices are actually true. So the correct response is to select nothing. But now that I think about this, we wouldn't know the difference between the student skipping the question and answering it. So in practice this is probably never used. There would have to be a choice like "none of the answers are true". 

So based on this I think you are right there isn't a valid use case where an answer is scored and there are no correct answers.

Reply:
Ah, I understand it now. Right, I think being explicit and providing "none of the answers are true" option makes more sense.